cypress/integration/v2/features/dynamic-content/latest-post.spec.js

```diff
@@ -0,0 +1,349 @@
+
+/**
+ * External dependencies
+ */
+import { registerTests } from '~stackable-e2e/helpers'
+import {
+	first, range,
+} from 'lodash'
+
+describe( 'Dynamic Content - Latest Post', registerTests( [
+	matchPostData,
+	adjustFieldOptions,
+	adjustFieldValues,
+] ) )
+
+const adjustPostField = ( fieldName, fieldOptions = {}, idx ) => {
+	cy.adjustDynamicContent( 'ugb/cta', 0, '.ugb-cta__title', {
+		source: 'Latest Post',
+		post: `${ idx }${ nth( idx ) } Latest Post`,
+		fieldName,
+		fieldOptions,
+	} )
+}
+
+/**
+ * Setup function.
+ *
+ * This is where we populate `@fieldsToAssert` for
+ * assertions.
+ *
+ * Prefetch field values and store it as alias.
+ *
+ * @example `@fieldsToAssert` structure
+ * ```
+ * [
+ * 		{
+ * 			name: <fieldName>
+ * 			value: <will assert value>,
+ * 			options: { ...<`adjustDynamicContent` command options> }
+ * 		}
+ * ]
+ * ```
+ *
+ * @param {number} idx
+ */
+function setFieldValues( idx ) {
+	cy.newPost()
+	// Define the alias.
+	cy.wrap( [] ).as( `fieldsToAssert${ idx }` )
+
+	// Populate Post Title.
+	cy.typePostTitle( `Latest Post #${ idx }` )
+	addToTest( { name: 'Post Title', value: `Latest Post #${ idx }` }, idx )
+
+	// Populate Post Slug.
+	addToTest( { name: 'Post Slug', value: `latest-post-${ idx }` }, idx )
+
+	// Populate Post Excerpt.
+	cy.addPostExcerpt( `Test excerpt #${ idx }` )
+	addToTest( { name: 'Post Excerpt', value: `Test excerpt #${ idx }` }, idx )
+
+	// Add a featured image.
+	cy.addFeaturedImage()
```

Comment:
we need to change the adding of featured Image. It should not register a post before adding featured image